internal/util/path_test.go

```diff
@@ -0,0 +1,68 @@
+package util
+
+import "testing"
+
+func TestJoinKey(t *testing.T) {
+	type args struct {
+		root string
+		k    string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "empty root",
+			args: args{root: "", k: "foo"},
+			want: "foo",
+		},
+		{
+			name: "empty key",
+			args: args{root: "foo", k: ""},
+			want: "foo",
+		},
+		{
+			name: "simple key",
+			args: args{root: "foo", k: "bar"},
+			want: "foo.bar",
+		},
+		{
+			name: "key with dot",
+			args: args{root: "foo", k: "bar.baz"},
+			want: "foo[\"bar.baz\"]",
```

Comment:
nit: This would read a little clearer if you use a raw string
```suggestion
			want: `foo["bar.baz"]`,
```